src/main/java/com/sap/sgs/phosphor/fosstars/tool/github/GitHubDataFetcher.java

```diff
@@ -0,0 +1,166 @@
+package com.sap.sgs.phosphor.fosstars.tool.github;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import org.apache.commons.collections4.map.LRUMap;
+import org.kohsuke.github.GHCommit;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+/**
+ * Helper class for GitHub data providers which pulls the data from a GitHub repository. Also, the
+ * class caches the fetched data for a certain amount of time.
+ */
+public class GitHubDataFetcher {
+
+  /**
+   * The singleton instance.
+   */
+  private static GitHubDataFetcher INSTANCE;
```

Comment:
Since the instance doesn't need `GitHub` in the constructor, let's make the field final.

```java
private static final GitHubDataFetcher INSTANCE = new GitHubDataFetcher();
```

Reply:
Agree 👍